### DISCUSSION TITLE
Wait until transaction occurred and is being confirmed

### DISCUSSION THREAD

**Discussion by AlbertoSinigaglia** [smartcontractkit/full-blockchain-solidity-course-py#17] [General]
Hi, during the video you are using things like
```
transaction.wait(1)
```
that in theory is pretty much like
```
import time
time.sleep(1)
```
however this is a "hope it finish"... in order to avoid this we might do something like this:
```
while transaction.confirmations < 1: 
    # in theory it should be 6/7 for Vitalik Buterin to reach a number of consensus to trust, 
    # but if we use brownie we have a single host
    time.sleep(1)
```

I've looked into `brownie` documentation and there is no such thing as "event" to wait for this to happen... is this the better way that we can come up to solve this problem (without busy waiting) or are there smarter ways to do that?

However, I don't understand why we are not `waint(1)` in the tests (4:52:09)... this is unsafe in case of a deployment in a test net not in local

**Comment by PatrickAlphaC** [comment]
Hmmm... I'm not quite following. Are you sure the timestamp is 4:52:09? I don't see a `wait` at that time.

**Reply by AlbertoSinigaglia** [reply]
My bad, i meant 4:42:09

**Reply by PatrickAlphaC** [reply]
Are you sure? https://youtu.be/M576WGiDBdQ?t=16931

**Reply by AlbertoSinigaglia** [reply]
I really don't know what's going on with youtube, I swear i was looking at the desired frame...
Try with this [https://youtu.be/M576WGiDBdQ?t=17161](https://youtu.be/M576WGiDBdQ?t=17161)

**Reply by AlbertoSinigaglia** [reply]
nevermind, I thought that the parameter of `wait` was seconds, but are confirmations after the transaction occurs, so there is no need for the cycle

**Comment by AlbertoSinigaglia** [comment]
nvm, `.wait(n)` waits already for n confirmations
